array/DNAm/preprocessing/checkRconfigFile.r

```diff
@@ -0,0 +1,164 @@
+##---------------------------------------------------------------------#
+##
+## Title: Check config.r 
+##
+## Purpose of script: Check config file parameters are present and in correct format
+##                    and if not exit the script with error message
+##
+##---------------------------------------------------------------------#
+
+
+#----------------------------------------------------------------------#
+# NOTES
+#----------------------------------------------------------------------#
+
+# Project Folder is provided on the command line
+
+
+#----------------------------------------------------------------------#
+# DEFINE PARAMETERS
+#----------------------------------------------------------------------#
+
+print("checking config.r file parameters are present and correctly formatted...")
+
+'%ni%' <- Negate('%in%') # define '%ni%' (not in)
+
+args<-commandArgs(trailingOnly = TRUE)
+dataDir <- args[1]
+configFile <- paste0(dataDir, "/config.r")
+
+source(configFile)
+
+
+# group config variables by type
+
+qcRmdParams <- c("projectTitle",
+                 "processedBy")           # strings
+
+
+
+qcthres <- c("thresBS",
+              "intenThres",
+              "nvThres",
+              "perMiss") # numeric
+
+
+logicalParams <- c("sexCheck",
+                   "snpCheck",
+                   "ctCheck")             # logical
+
+
+
+# ungrouped params
+ungrouped <- c("tissueType", "arrayType", "techVar", "bioVar")
+
+
+
+# only used if ctCheck = TRUE
+ctThres <- c("studentThres",         
+             "nSDThres")                  # numeric
+
+ctCellParams <- c("predDistinctCT",               
+                  "neunCT")               # strings 
+
+
+
+#----------------------------------------------------------------------#
+# Check config file parameters
+#----------------------------------------------------------------------#
+
+
+# check necessary parameters exist
+
+for(i in c(qcRmdParams, qcthres, logicalParams, ungrouped)){
+  if(!exists(i)){
+    stop(paste0("'", i, "' must be defined in the config file" ))

```

Comment:
There is a small problem with using `stop` here. The user will have to run this script checker way more times than necessary as the script stops checking once it finds a single error. If there are 5 errors, then the user will have to run this script 6 times minimum.

Reply:
good point - corrected in #209